main.cpp

```diff
@@ -298,12 +205,30 @@ struct FloatType
     FloatType& multiply(float rhs );
     FloatType& divide(float rhs );
 
-    operator float() { return *value; }
+    FloatType& pow(const FloatType& rhs);
+    FloatType& pow(const DoubleType& rhs);
+    FloatType& pow(const IntType& rhs);
+    FloatType& pow(float rhs);
+
+    operator float() const { return *value; }
 
 private:
     float* value;
+    FloatType& powInternal(float arg);
 };
 
+FloatType& FloatType::pow(float rhs)
+{
+    powInternal(rhs);
+    return *this;
+}
```

Comment:
what does `powInternal` return?
return the result of calling `powInternal`
```
return powInternal()
```
- [ ] revise